source/_less/position.less

```diff
@@ -0,0 +1,198 @@
+@import 'vars';
+
+@popover-background-color: @gray-0;
+@popover-border-color: @gray-300;
+@popover-shadow-color: fade(@gray-900, 16%);
+@popover-arrow-width: 7px;
+@popover-arrow-outer-width: 8px;
+
+:host {
+  #arrow {
+    &,
+    &:after {
+      position: absolute;
+      display: block;
+      width: 0;
+      height: 0;
+      border-color: transparent;
+      border-style: solid;
+    }
+  }
+
+  #arrow {
+    border-width: @popover-arrow-outer-width;
+    background-color: @gray-0;
+    transform: rotate(-45deg);
+  }
+
+  #arrow:after {
+    border-width: @popover-arrow-width;
+    background-color: @gray-0;
+    content: " ";
+  }
+}
+
+:host {
+  &([position='top']) #arrow {
+    bottom: -8px;
+    left: 50%;
+    margin-left: -8px;
+    box-shadow: -3px 3px 3px 0 @popover-shadow-color;
+    border-color: transparent transparent @popover-border-color @popover-border-color;
+
+    &::after {
```

Comment:
What's the point of `:after` if you have a box-shadow?

Reply:
The `box-shadow` by itself adds a little bit of shadow visibility where we don't want it. Here's how it looks without `:after`:

![screen shot 2017-12-01 at 10 32 35 am](https://user-images.githubusercontent.com/6867918/33492606-2be6bb76-d683-11e7-84dc-15bf8b6f293f.png)
_Small bit of shadow peaking out on the left_

In the `:after` styles, the inner portion of the arrows are covered in white to prevent this from happening.

Reply:
What if we, use `::before` to apply a box-shadow _behind_ the tooltip, and the `::after` to apply the color _in front_ of the tooltip using a hard linear gradient?

_layers (from top to bottom)_
* `::after` Arrow background/color (see [this codepen prototype](https://codepen.io/CITguy/pen/b9ee296305c336b39f2b77a60da4c6d0))
* `div` Tooltip container
* `::before` Arrow box-shadow